test/run/variant-specials-bug.mo

```diff
@@ -0,0 +1,13 @@
+func specials(two : { #c0; #c1 }) {
+  switch two {
+    case (#c2) ();           // note #c2 <> #c1
+    case (#c2) assert false; // note #c2 <> #c1
+  };
```

Comment:
now that this traps (`pattern failed`), it won't get to the latter switch. I'll split the test.